metaserver/src/main/scala/scala/meta/languageserver/search/InMemorySymbolIndex.scala

```diff
@@ -185,4 +188,38 @@ class InMemorySymbolIndex(
     Effects.IndexSemanticdb
   }
 
+  private def hasOneOfFlag(flags: Long, flag: Long): Boolean =
+    (flags & flag) != 0L
+
+  override def workspaceSymbols(query: String): List[SymbolInformation] = {
+    import scala.meta.semanticdb._
+    import scala.meta.languageserver.ScalametaEnrichments._
+    val result = symbolIndexer.allSymbols.toIterator
```

Comment:
I expected this to be prohibitively expensive but for scalameta/language-server it's still quite snappy. I'm sure for larger projects it may lag a bit. However, once we move to a sql-based index this should return instantly even for massive projects.